distance_strike_feature.py

```diff
@@ -0,0 +1,374 @@
+        """
+        Computes the distance strikes percentages for each fighter in the dataset
+
+        Parameters:
+        - fighter_a_distance_strikes_hit (float): The number of distance strikes hit by fighter a
+        - fighter_a_distance_strikes_taken (float): The number of distance strikes taken by fighter a
+        - fighter_b_distance_strikes_hit (float): The number of distance strikes hit by fighter b
+        - fighter_b_distance_strikes_taken (float): The number of distance strikes taken by fighter b
+        - differential (bool): Whether to compute the differential or not
+
+        Returns:
+        - (float, float): The distance strikes percentages for each fighter as a tuple
+        """
+        fighter_a_distance_strikes = fighter_a_distance_strikes_hit / fighter_a_distance_strikes_taken if fighter_a_distance_strikes_taken != 0 else 0
+        fighter_b_distance_strikes = fighter_b_distance_strikes_hit / fighter_b_distance_strikes_taken if fighter_b_distance_strikes_taken != 0 else 0
+
+        if differential:
+            return fighter_a_distance_strikes - fighter_b_distance_strikes, fighter_b_distance_strikes - fighter_a_distance_strikes
+        else:
+            return fighter_a_distance_strikes, fighter_b_distance_strikes
+    
+    def compute_distance_strikes(self, df, fighter_a_id, fighter_b_id, index, round_number, last_fights, column_name, differential=False, is_defense=False, is_percentage=False):
+        """
+        Computes the distance strikes for each fighter in the dataset based on column_name
+
+        Parameters:
+        - df (pd.Dataframe): The original dataframe containing all the fights
+        - fighter_a_id (int): The id of the first fighter
+        - fighter_b_id (int): The id of the second fighter
+        - index (int): The index of the current fight
+        - round_number (int): The round number of the current fight
+        - last_fights (int): The number of last fights to consider
+        - column_name (string): The name of the column to compute
+        - differential (bool): Whether to compute the differential or not
+        - is_defense (bool): Whether to compute the defense or not
+        - is_percentage (bool): Whether to compute the percentage or not
+
+        Returns:
+        - (float, float): The distance strikes for each fighter as a tuple
+        """
+        all_prev_fights = df.loc[:index-1]
+        
+        if not all_prev_fights.empty:
+            fighter_a_id_vals = all_prev_fights.fighter_a_id.values
+            fighter_b_id_vals = all_prev_fights.fighter_b_id.values
+
+            fighter_a_fights = all_prev_fights[(fighter_a_id_vals == fighter_a_id)][-last_fights:]
+            fighter_b_fights = all_prev_fights[(fighter_b_id_vals == fighter_b_id)][-last_fights:]
+
```

Comment:
@ivan-guo I think this is only getting the fights for each fighter where they only fighter_a or figher_b. as in only returning the fights where fighter_a is fighter a and fighter_b is only fighter_b. Not considering the case where they are the other fighter. 

in get_fighter_distance_details it does looks for where the fighter is either a or b but that function gets all past fights for the fighter form this function so it would only have the fights where fighter_a is fighter_a

Reply:
Oh shit I totally missed this - good catch. Yes this should be 

`fighter_a_fights = all_prev_fights[(fighter_a_id_vals == fighter_a_id) | (fighter_b_id_vals == fighter_a_id)][-last_fights:]`

See https://github.com/ivan-guo/OctaPicks/blob/f7ca581e2f424b99758d133b3dfdad8d170c8b31/knockdowns.py#L67

Reply:
i think this is the case with the other pr's too

Reply:
right shoot missed that case here, fixed it in the ground strike feature and clinch strike too. Thanks!